python/gym_ignition/randomizers/base/model.py

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2020 Istituto Italiano di Tecnologia (IIT). All rights reserved.
+# This software may be modified and distributed under the terms of the
+# GNU Lesser General Public License v2.1 or any later version.
+
+import abc
+
+
+class ModelRandomizer(abc.ABC):
+
+    @abc.abstractmethod
+    def randomize_model(self) -> str:
+        """
+        Randomize the model.
```

Comment:
In the future documentation it would be very useful to have a list of all the randomizable quantities.

Reply:
Given that you can access all the elements through XPath patterns, you can randomize everything in the SDF. The only constraint is that randomized quantities are elements and not attributes. This was a design choice that matches SDF, in fact all relevant data are store in elements.

Reply:
Ok, I see.

Reply:
Users may find it useful to refer to a set of examples. The cartpole ones are a great point to start. In the future, it may be valuable to also provide an example involving a more complex robot like Panda or iCub.

Reply:
This PR introduces a test with a Panda:

https://github.com/robotology/gym-ignition/blob/b1ff4419743c71e49e457b42213170a6c0f2f415/tests/test_gym_ignition/test_sdf_randomizer.py#L215

 In general, tests are a great piece of implicit documentation that many ignore.